src/components/MAPI/organizations/utils.ts

```diff
@@ -64,3 +72,101 @@ export function computeClusterCountersForOrganizations(clusters?: Cluster[]) {
     return acc;
   }, {});
 }
+
+export async function fetchClusterListForOrganizations(
+  httpClientFactory: HttpClientFactory,
+  auth: IOAuth2Provider,
+  cache: Cache,
+  provider: PropertiesOf<typeof Providers>,
+  organizations: Record<string, IOrganization>
+): Promise<ClusterList> {
+  const request: authorizationv1.ISelfSubjectAccessReviewSpec = {
+    resourceAttributes: {
+      namespace: '',
+      verb: 'list',
+      group: 'cluster.x-k8s.io',
+      resource: 'clusters',
+    },
+  };
+
+  const accessReviewResponse =
+    await authorizationv1.createSelfSubjectAccessReview(
+      httpClientFactory(),
+      auth,
+      request
+    );
+
+  if (accessReviewResponse.status?.allowed) {
+    // If the user has access to list clusters at the cluster scope,
+    // return the function to list clusters at the cluster scope
+    return fetchClusterList(httpClientFactory, auth, provider, '');
+  }
+
+  const clusterList: capiv1alpha3.IClusterList = {
+    apiVersion: 'cluster.x-k8s.io/v1alpha3',
+    kind: capiv1alpha3.ClusterList,
+    metadata: {},
+    items: [],
+  };
+
+  const requests = Object.entries(organizations).map(
+    async ([organizationName, organizationEntry]) => {
+      const clusterListKey = fetchClusterListKey(
+        provider,
+        organizationEntry.namespace,
+        organizationName
+      );
+      const cachedClusterList: capiv1alpha3.IClusterList | undefined =
+        cache.get(clusterListKey);
+
```

Comment:
Trying to improve performance here as well by checking if there is already cached data in `SWR` for clusters for a specific namespace, before launching another fetch request. If data is fetched, then we write it to the cache. This is similar to how we check for `appcatalogentries` in the cache before fetching.